docs/data/material/guides/migration-v4/migration-v4.md

```diff
@@ -2949,8 +2948,9 @@ and [an explicit name for the stylesheet](https://github.com/garronej/tss-react#
 
 function Nested(props) {
 - const classes = useStyles(props);
-+ let { classes } = useStyles();
-+ classes = useMergedClasses(classes, props.classes);
++ const { classes } = useStyles(undefined, { props });
```

Comment:
Just curious, what's the `undefined` as the first param?

Reply:
Hi @mnajdova,  
The first argument is for the parameters needed for computing the internal styles,  
the second one is:  
1) For merging `classes`
2) For defining what will be passed to the [mui styleOverrides callback](https://mui.com/customization/theme-components/#overrides-based-on-props).   

```typescript
import { useReducer } from "react";
import { makeStyles } from "tss-react/mui";


export type Props = {
	className?: string;
	classes?: Partial<ReturnType<typeof useStyles>["classes"]>;
	lightBulbBorderColor: string;
}

export function MyComponent(props: Props) {

	const { className } = props;

	const [isOn, toggleIsOn] = useReducer(isOn => !isOn, false);

	const { classes, cx } = useStyles({ isOn }, { props, "ownerState:" { isOn } });

	return (
		<div className={cx(classes.root, className)} >
			<div className={classes.lightBulb}></div>
			<button onClick={toggleIsOn}>{`Turn ${isOn ? "off" : "on"}`}</button>
		</div>
	);

}

const useStyles = makeStyles({ "name": "MyComponent" })<{ isOn: boolean; }>()(
	(theme, { isOn }) => ({
		"root": {
			"padding": theme.spacing(2)
		},
		"lightBulb": {
			"backgroundColor": isOn ? "yellow" : "grey"
		}
	})
);
```

When the internal style doesn’t need any parameter to be computed then the first argument is `undefined`.  

Best